tests/common.go

```diff
@@ -5311,3 +5318,122 @@ func IsEksPxOperator() bool {
+//	executionMode: The mode to use for executing the operation, either "Sequential" or "Parallel".
+//
+// # Example
+//
+// The original code:
+//
+//	for _, value := range items /*slice, array or map*/ {
+//		operation(value)
+//	}
+//
+// or
+//
+//	for index, value := range items /*slice or array*/ {
+//		operation(index, value)
+//	}
+//
+// or
+//
+//	for key, value := range items /*map*/ {
+//		operation(key, value)
+//	}
+//
+// The original code uses a common pattern for iterating over a slice, array or map of items and calling the 'operation'
+// function for each item. To simplify this pattern and allow for concurrent execution of the 'operation'
+// function you can replace the for loops with a call to ConcurrentExecute(items, operation, executionMode), where
+// 'executionMode' is either 'Parallel' or 'Sequential'.
+func ConcurrentExecute(items interface{}, operation interface{}, executionMode ExecutionMode) error {
+	v := reflect.ValueOf(items)
+	var keys []reflect.Value
+	if v.Kind() == reflect.Map {
+		keys = v.MapKeys()
+	} else if v.Kind() == reflect.Slice || v.Kind() == reflect.Array {
+		keys = make([]reflect.Value, v.Len())
+		for i := 0; i < v.Len(); i++ {
+			keys[i] = v.Index(i)
+		}
+	} else {
+		return fmt.Errorf("instead of %#v, items should be a slice, array, or map", v.Kind().String())
+	}
+
+	length := len(keys)
+	if length == 0 {
+		return nil
+	} else if length == 1 {
+		executionMode = Sequential
+	}
+
+	fnValue := reflect.ValueOf(operation)
+	numArgs := fnValue.Type().NumIn()
+
+	callOperation := func(key reflect.Value) {
+		in := make([]reflect.Value, numArgs)
+		in[0] = key
+		if numArgs == 2 {
+			in[1] = v.MapIndex(key)
+		}
+		fnValue.Call(in)
+	}
+
+	if executionMode == Sequential {
+		for i := 0; i < length; i++ {
+			callOperation(keys[i])
+		}
+	} else {
+		var wg sync.WaitGroup
+		for i := 0; i < length; i++ {
+			wg.Add(1)
+			go func(i int) {
+				defer wg.Done()
+				callOperation(keys[i])
+			}(i)
+		}
+		wg.Wait()
+	}
+
+	return nil
+}
+
+// RandomUniqueStrings selects a random subset of unique strings from the input array,
+// with the given length. It returns a new array with the selected strings in random order.
+// If length is zero or negative or greater than the length of the input array, it also returns an error.
+//
+// Parameters:
+// - strings: a array of strings to select from.
+// - length: the number of strings to select from the input array.
+//
+// Returns:
+// - a new slice of strings with the selected strings in random order.
+// - an array if the length parameter is zero or negative, or if it is greater than the length of the input array.
+func RandomUniqueStrings(strings []string, length int) ([]string, error) {
+	if length <= 0 {
+		return nil, fmt.Errorf("length must be greater than zero")
+	}
+	if length > len(strings) {
+		return nil, fmt.Errorf("length cannot be greater than the length of the input strings")
+	}
+	randomstrings := make([]string, length)
```

Comment:
Make this camel case - `randomStrings`

Reply:
Replaced "randomstrings" with "randomStrings".